src/app/(auth)/signup/page.js

```diff
@@ -1,14 +1,33 @@
 "use client";
 
-import { useState } from "react";
+import { useState, useEffect } from "react";
+import { useRouter } from "next/navigation";
 import SignUpAccount from "./components/SignUpAccount";
 import SignUpDetails from "./components/SignUpDetails";
 import SignUpEmailConfirmation from "./components/SignUpEmailConfirmation";
 import SignUpChildren from "./components/SignUpChildren";
+import { AUTH_TOKEN_NAME } from "@/app/api.mjs";
 
 export default function SignUp() {
+  const router = useRouter();
   const [currentStep, setCurrentStep] = useState(0);
+
+  useEffect(() => {
+    const authToken = localStorage.getItem(AUTH_TOKEN_NAME);
+    if (authToken) {
+      router.push('/');
+    }
+    const savedStep = localStorage.getItem('signup_current_step');
+    setCurrentStep(savedStep ? parseInt(savedStep) : 0);
+  }, []);
+
+  useEffect(() => {
```

Comment:
So here is what we have:
scrolNext() -> state -> localStorage

From Python Zen: flat is better than nexted.
So I would update both current step state and local storage variable from scrollNext function.
This way we have
scrollNext() -> state and localStorage